File: 신고_결과_받기.py
def solution(id_list, reports, k):
    reports = set(reports)
    n_reported = {user: 0 for user in id_list}
    reported_by = {user: [] for user in id_list}
    for report in reports:
        reporter, reported = report.split()
        n_reported[reported] += 1
        reported_by[reporter].append(reported)
    
    n_emails = []
    for user in id_list:
        cnt = 0
        for reported in reported_by[user]:
            if n_reported[reported] >= k:
                cnt += 1
        n_emails.append(cnt)
        
    return n_emails
            

File: golang/Level 1/신고_결과_받기.go
import (
    "strings"
)

func removeDuplicateStr(strSlice []string) []string {
    allKeys := make(map[string]bool)
    list := []string{}
    for _, item := range strSlice {
        if _, value := allKeys[item]; !value {
            allKeys[item] = true
            list = append(list, item)
        }
    }
    return list
}

func solution(id_list []string, reports []string, k int) []int {
    reports = removeDuplicateStr(reports)
    
    reported_by := make(map[string]map[string]bool)
    for _, id := range id_list {
        reported_by[id] = make(map[string]bool)
    }
    
    n_reports := make(map[string]int)
    for _, id := range id_list {
        n_reports[id] = 0
    }
    
    for _, report := range reports {
        slice := strings.Split(report, " ")
        reporter := slice[0]
        reported := slice[1]
        
        n_reports[reported] += 1
        reported_by[reporter][reported] = true
    }
    
    n_emails := make([]int, len(id_list))
    for i, user := range id_list {
        for reported, _ := range reported_by[user] {
            if n_reports[reported] >= k {
                n_emails[i]++
            }
        }
    }
    
    return n_emails
}